### ISSUE TITLE
Feed auto-discovery on websites does not always work

### ISSUE THREAD

**Issue by kvibber** [gugray/rss-parrot#27]
Every time I've entered a website address that I know has `<link>`-based autodiscovery, I've gotten the "Hm, I can't find a feed for this site." response from the bot. Some examples:

https://wordpress.org/news/ (and every other WordPress site I've tried)
https://kvibber.com/

It has no problem if I ask it for the feed URLs, but it's not finding the `<link rel="alternate"...` feeds for some reason.

**Comment by gugray** [comment]
With kvibber.com, the request timed out in 10 seconds. I've seen this with a few sites; it looks like there is something between the parrot's datacenter and the website that blocking the request. I'm not sure how to diagnose this further or what I may be able to change in the parrot so it's blocked less often...

Any info you can share about kvibber.com's hosting that might help here?

**Comment by kvibber** [comment]
Odd... it's a 12KB static html file (4KB compressed) served by Apache on a Dreamhost VPS with a Let's Encrypt TLS certificate. I can't think why it would time out unless the server was overloaded by something else at the time.

And the replies from birb have been fast - within a second. I tested it again while tailing the website server logs, and I got the following line showing up immediately after hitting Publish on the Mastodon DM:

```
64.176.202.191 - - [11/Jan/2024:08:49:30 -0800] "GET / HTTP/2.0" 200 4364 "-" "RSS-Parrot-Bot/0.0.44 (+https://rss-parrot.net)" 
```

followed just as quickly with the reply from birb saying it couldn't find a feed.

So I don't think timeouts are the only issue.

Especially since I see the same issue when asking birb for other website URLs like

https://isc.sans.edu/
https://news.indieweb.org/en
https://www.11ty.dev/blog/

and so on.
